### ISSUE TITLE
Where can I put this makeRecord function?

### ISSUE THREAD

**Issue by ocharles** [dhall-lang/dhall-haskell#337]
In `dhall-to-cabal`, I built an applicative functor for parsing records: https://github.com/dhall-lang/dhall-to-cabal/blob/master/lib/Dhall/Extra.hs#L36,L84. I've already needed this in another project, so it'd be nice to have it somewhere.

This is useful when you have to parse a Dhall record to a Haskell value (usually a Haskell record), but can't use generics. Without this, you have to be careful when constructing a `Type` that the `expected` type matches what you want to `extract`. With `RecordBuilder`, this plumbing is taken care of for you.

Do you want this in dhall-haskell, or should we look at dhall-haskell-contrib?

**Comment by ocharles** [comment]
The dual to this is also in `dhall-to-cabal`: https://github.com/dhall-lang/dhall-to-cabal/blob/master/cabal-to-dhall/Main.hs#L183,L207

**Comment by Gabriel439** [comment]
Yeah, you can definitely add these to the `dhall` library.  I think the most appropriate place to put them would be in the main `Dhall` module since that contains all the `Type`-related logic

**Comment by ocharles** [comment]
Great, I'll package this all up into a PR.

**Comment by ocharles** [comment]
Yep
